mapbox/libandroid-telemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -0,0 +1,573 @@
+import com.mapbox.services.android.telemetry.constants.TelemetryConstants;
+import com.mapbox.services.android.telemetry.http.TelemetryClient;
+import com.mapbox.services.android.telemetry.location.AndroidLocationEngine;
+import com.mapbox.services.android.telemetry.location.LocationEngine;
+import com.mapbox.services.android.telemetry.location.LocationEngineListener;
+import com.mapbox.services.android.telemetry.permissions.PermissionsManager;
+import com.mapbox.services.android.telemetry.service.TelemetryService;
+import com.mapbox.services.android.telemetry.utils.TelemetryUtils;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.util.Hashtable;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.UUID;
+import java.util.Vector;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Response;
+import okhttp3.internal.Util;
+import timber.log.Timber;
+
+/**
+ * This is the entry point to manage Mapbox telemetry
+ */
+public class MapboxTelemetry implements Callback, LocationEngineListener {
+
+  private static MapboxTelemetry instance;
+
+  private boolean initialized = false;
+  private Context context = null;
+  private String accessToken = null;
+  private String mapboxSessionId = null;
+  private long mapboxSessionIdLastSet = 0;
+  private String mapboxVendorId = null;
+  private DisplayMetrics displayMetrics = null;
+  private Intent batteryStatus = null;
+  private TelemetryClient client = null;
+  private Vector<Hashtable<String, Object>> events = new Vector<>();
+  private Timer timer = null;
+  private LocationEngine locationEngine = null;
+  private boolean withShutDown = false;
+  private Boolean telemetryEnabled = null;
+
+  /**
+   * Private constructor for configuring the single instance per app.
+   */
+  private MapboxTelemetry() {
+  }
+
+  /**
+   * Primary access method (using singleton pattern)
+   *
+   * @return MapboxTelemetry
+   */
+  public static MapboxTelemetry getInstance() {
+    if (instance == null) {
+      instance = new MapboxTelemetry();
+    }
+
+    return instance;
+  }
+
+  /**
+   * Initialize MapboxTelemetry.
+   *
+   * @param context        The context associated with the application
+   * @param accessToken    The accessToken associated with the application
+   * @param locationEngine Initialize telemetry with a custom location engine
+   */
+  public void initialize(@NonNull Context context, @NonNull String accessToken,
+                         @NonNull LocationEngine locationEngine) {
+    this.locationEngine = locationEngine;
+    initialize(context, accessToken);
+  }
+
+  /**
+   * Initialize MapboxTelemetry.
+   *
+   * @param context     The context associated with the application
+   * @param accessToken The accessToken associated with the application
+   */
+  public void initialize(@NonNull Context context, @NonNull String accessToken) {
+    if (initialized) {
+      return;
+    }
+
+    Timber.v("Initializing telemetry.");
```

Comment:
Right. Notice all this log calls are `.v` as in verbose. These can easily be filtered out by the right Timber `Tree`. Either way, we can remove these _after_ the SDK happens and we're sure  the refactor is working as expected.